File: Q3_Cpp/src/Product/ComputerGame/ComputerGame.cpp
#include "../../../include/Product/ComputerGame/ComputerGame.h"

// Constructor for ComputerGame
ComputerGame::ComputerGame(std::string id, std::string name, Supplier supplier, int packagingUnits, double price, std::string developer, int ageRating)
    : Product(id, name, supplier, packagingUnits, price), developer(developer), ageRating(ageRating) {
}

// Getter and setter for developer
std::string ComputerGame::getDeveloper() const {
    return developer;
}

void ComputerGame::setDeveloper(const std::string& developer) {
    this->developer = developer;
}

// Getter and setter for ageRating
int ComputerGame::getAgeRating() const {
    return ageRating;
}

void ComputerGame::setAgeRating(int ageRating) {
    this->ageRating = ageRating;
}

File: Q1_Java/src/main/java/Product/ComputerGame/ComputerGame.java
package Product.ComputerGame; // Package for ComputerGame

import Product.Product;
import Product.Supplier;

// Abstract class ComputerGame, extends Product
public abstract class ComputerGame extends Product {
    private String developer; // Developer of the computer game
    private int ageRating; // Age rating for the game

    // Constructor for ComputerGame
    public ComputerGame(String id, String name, Supplier supplier, int packagingUnits, double price, String developer, int ageRating) {
        super(id, name, supplier, packagingUnits, price); // Calling Product's constructor
        this.developer = developer; // Setting the developer
        this.ageRating = ageRating; // Setting the age rating
    }

    // Getter and setter for developer
    public String getDeveloper() { return developer; }
    public void setDeveloper(String developer) { this.developer = developer; }

    // Getter and setter for ageRating
    public int getAgeRating() { return ageRating; }
    public void setAgeRating(int ageRating) { this.ageRating = ageRating; }
}
